File: src/Python/greedy/121.买卖股票的最佳时机.py
#
# @lc app=leetcode.cn id=121 lang=python
#
# [121] 买卖股票的最佳时机
#

# @lc code=start
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        maxPro = 0
        minPri = 10000
        for price in prices:
            minPri = min(minPri, price)
            maxPro = max(maxPro, price - minPri)
        return maxPro
# @lc code=end

File: src/Javascript/greedy/121.买卖股票的最佳时机.js
/*
 * @lc app=leetcode.cn id=121 lang=javascript
 *
 * [121] 买卖股票的最佳时机
 */

// @lc code=start
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxPro = 0, minInput = 10000;
    for (let price of prices) {
        minInput = Math.min(price, minInput);
        maxPro = Math.max(maxPro, (price - minInput));
    }
    return maxPro;
};
// @lc code=end

